src/lib/functions/registry.ts

```diff
@@ -527,7 +550,6 @@ export class FunctionsRegistry {
         const func = new NetlifyFunction({
           blobsContext: this.blobsContext,
           config: this.config,
-          directory: directories.find((directory) => mainFile.startsWith(directory)),
```

Comment:
This was weird. We were computing `directory` by looking at the function and ascertaining which one of the main directories it belonged to. We then passed this property along and [further downstream](https://github.com/netlify/cli/pull/7408/files#diff-3177d45a0f449e1b37a03d73a574731300452cc8ddf10c52b2bbae400371d7fdL75) we'd use it to decide what path to send to `zipFunction`, which should always be the path to the function boundary (i.e. the function sub-directory if it lives in a sub-directory, or the function main file if not).

But that's exactly what the `srcPath` that we get from zip-it-and-ship-it is. So I've removed this property altogether, passed `srcPath` to `NetlifyFunction` and then read it from there when calling `zipFunction`.